tests/scripts_run/test_main.py

```diff
@@ -0,0 +1,774 @@
+"""Tests for the main() function and end-to-end execution workflows in spd/scripts/run.py.
+
+This file focuses on testing the high-level integration behavior of the main() function,
+including local execution, SLURM submission, and end-to-end parameter sweep workflows.
+
+Lower-level functions like generate_grid_combinations and load_sweep_params are tested
+in test_grid_search.py and test_run_sweep_params.py respectively.
+"""
+
+# pyright: reportUnknownParameterType=false, reportMissingParameterType=false, reportUnusedParameter=false
+
+import json
+from unittest.mock import Mock, patch
+
+import pytest
+
+from spd.scripts.run import (
+    generate_commands,
+    generate_run_id,
+    main,
+    resolve_sweep_params_path,
+)
+
+
+def get_valid_tms_config():
+    """Get a valid TMS experiment config."""
+    return {
```

Comment:
I think I'd prefer to just read one of our canonical configs (`spd/experiments/tms/tms_5-2_config.yaml`). Main reason is that it's very annoying to update all of configs in tests when something in the code changes.